packages/hash/api/src/model/knowledge/account.fields.ts

```diff
@@ -0,0 +1,77 @@
+import { OrgModel, UserModel } from "..";
+import { GraphApi } from "../../graph";
+import {
+  generateSchemaBaseUri,
+  generateWorkspacePropertyTypeSchema,
+  RESTRICTED_SHORTNAMES,
+  workspaceTypesNamespaceUri,
+} from "../util";
+
+module AccountId {
+  // Generate the schema for the account id property type
+  export const accountIdPropertyType = generateWorkspacePropertyTypeSchema({
+    title: "Account ID",
+    possibleValues: [{ primitiveDataType: "Text" }],
+  });
+
+  export const accountIdBaseUri = generateSchemaBaseUri({
+    namespaceUri: workspaceTypesNamespaceUri,
+    kind: "propertyType",
+    title: accountIdPropertyType.title,
+  });
+}
+
+module Shortname {
+  // Generate the schema for the shortname property type
+  export const shortnamePropertyType = generateWorkspacePropertyTypeSchema({
+    title: "Shortname",
+    possibleValues: [{ primitiveDataType: "Text" }],
+  });
+
+  export const shortnameBaseUri = generateSchemaBaseUri({
+    namespaceUri: workspaceTypesNamespaceUri,
+    kind: "propertyType",
+    title: shortnamePropertyType.title,
+  });
+
+  // Validations for shortnames
+  export const shortnameMinimumLength = 4;
+  export const shortnameMaximumLength = 24;
+
+  const ALLOWED_SHORTNAME_CHARS = /^[a-zA-Z0-9-_]+$/;
+
+  export const shortnameContainsInvalidCharacter = (
+    shortname: string,
+  ): boolean => {
+    return !!shortname.search(ALLOWED_SHORTNAME_CHARS);
+  };
+
+  export const shortnameIsRestricted = (shortname: string): boolean => {
+    return RESTRICTED_SHORTNAMES.includes(shortname);
+  };
+
+  export const shortnameIsTaken = async (
+    graphApi: GraphApi,
+    params: { shortname: string },
+  ) => {
+    return (
+      (await UserModel.getUserByShortname(graphApi, params)) !== null ||
```

Comment:
This validation introduces a circular dependency. Not sure what we can do about it, as we need to define the validation somewhere and use it from the model classes themselves

Reply:
If this isn't causing compilation errors then perhaps we can throw a todo in and revisit later. I think it's becoming pretty apparent we need to rethink the model classes (hence the asana task haha)

Reply:
Added todo here 6fe4d3894b0a142941ba3a309d8d76b13b1a450d